Userland/Applications/HexEditor/HexEditorWidget.cpp

```diff
@@ -226,6 +248,94 @@ HexEditorWidget::HexEditorWidget()
+    } else {
+        m_value_inspector_model->set_parsed_value(ValueInspectorModel::ValueType::SignedByte, "");
+        m_value_inspector_model->set_parsed_value(ValueInspectorModel::ValueType::UnsignedByte, "");
+    }
+
+    if (byte_read_count >= 2) {
+        u16 unsigned_short_value = 0;
+        if (m_value_inspector_little_endian)
+            unsigned_short_value = (unsigned_64_bit_int & 0xFFFF);
+        else
+            unsigned_short_value = (unsigned_64_bit_int >> (64 - 16)) & 0xFFFF;
+
+        m_value_inspector_model->set_parsed_value(ValueInspectorModel::ValueType::SignedShort, String::number((i16)unsigned_short_value));
+        m_value_inspector_model->set_parsed_value(ValueInspectorModel::ValueType::UnsignedShort, String::number(unsigned_short_value));
+    } else {
+        m_value_inspector_model->set_parsed_value(ValueInspectorModel::ValueType::SignedShort, "");
+        m_value_inspector_model->set_parsed_value(ValueInspectorModel::ValueType::UnsignedShort, "");
+    }
+
+    if (byte_read_count >= 4) {
+        u32 unsigned_int_value = 0;
+        if (m_value_inspector_little_endian)
+            unsigned_int_value = (unsigned_64_bit_int & 0xFFFFFFFF);
+        else
+            unsigned_int_value = (unsigned_64_bit_int >> 32) & 0xFFFFFFFF;
+
+        m_value_inspector_model->set_parsed_value(ValueInspectorModel::ValueType::SignedInt, String::number((i32)unsigned_int_value));
+        m_value_inspector_model->set_parsed_value(ValueInspectorModel::ValueType::UnsignedInt, String::number(unsigned_int_value));
+        m_value_inspector_model->set_parsed_value(ValueInspectorModel::ValueType::Float, String::number(bit_cast<float>(unsigned_int_value)));
+    } else {
+        m_value_inspector_model->set_parsed_value(ValueInspectorModel::ValueType::SignedInt, "");
+        m_value_inspector_model->set_parsed_value(ValueInspectorModel::ValueType::UnsignedInt, "");
+        m_value_inspector_model->set_parsed_value(ValueInspectorModel::ValueType::Float, "");
+    }
+
+    if (byte_read_count >= 8) {
+        m_value_inspector_model->set_parsed_value(ValueInspectorModel::ValueType::SignedLong, String::number((i64)(unsigned_64_bit_int)));
+        m_value_inspector_model->set_parsed_value(ValueInspectorModel::ValueType::UnsignedLong, String::number(unsigned_64_bit_int));
+        m_value_inspector_model->set_parsed_value(ValueInspectorModel::ValueType::Double, String::number(bit_cast<double>(unsigned_64_bit_int)));
+    } else {
+        m_value_inspector_model->set_parsed_value(ValueInspectorModel::ValueType::SignedLong, "");
+        m_value_inspector_model->set_parsed_value(ValueInspectorModel::ValueType::UnsignedLong, "");
+        m_value_inspector_model->set_parsed_value(ValueInspectorModel::ValueType::Double, "");
+    }
+
+    // TODO: Parse as other values like ASCII, UTF8, UTF16, Timestamp etc
```

Comment:
please use "FIXME" as per the style guide.

Reply:
Fixed